xgcm/test/test_grid.py

```diff
@@ -782,3 +782,26 @@ def test_boundary_kwarg_same_as_grid_constructor_kwarg():
     actual2 = grid2.interp(ds.data_g, ("X", "Y"))
 
     xr.testing.assert_identical(actual1, actual2)
+
+
+@pytest.mark.parametrize(
+    "metric_axes,metric_name",
+    [
+        ("X", "dx_t"),
+        ("Y", "dy_ne"),
+    ],
+)
+def test_interp_like(metric_axes, metric_name):
+
+    ds, coords, _ = datasets_grid_metric("C")
+
+    grid = Grid(ds, coords=coords)
+    grid.set_metrics(metric_axes, metric_name)
+    metric_available = grid._metrics.get(frozenset(metric_axes), None)
```

Comment:
Can you use one of the data variables from the test dataset ('u', 'v', or 'tracer') here instead of the coordinates? I think that makes the code smaller and easier to understand. Basically I would try to get rid of any 'metric' mention in this test. I can help with this.

Reply:
Do you mean something like:

```python
interp = grid.interp_like(ds.v, ds.u)
```

where ds.v is the "source" and ds.u is the "target"? I'm not sure how to construct the next statement for comparison since grid.interp works with an axis specification:

```python
test_metric = grid.interp(ds[metric_name], metric_axes)
```

Reply:
If you use the dictionary style notation (`ds['u']`) you can just define var_name='u' and then you just need to figure out the manual steps required to go from source to target. This will be a single or two axes that need to be interpolated, very similar to the other test.